pkg/apis/settings/settings_validation.go

```diff
@@ -0,0 +1,91 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package settings
+
+import (
+	"fmt"
+	"net/url"
+	"time"
+
+	"go.uber.org/multierr"
+	"knative.dev/pkg/apis"
+
+	"github.com/aws/karpenter/pkg/apis/v1alpha1"
+)
+
+func (s Settings) Validate() error {
+	return multierr.Combine(
+		s.validateEndpoint(),
+		s.validateTags(),
+		s.validateClusterName(),
+		s.validateVMMemoryOverheadPercent(),
+		s.validateReservedENIs(),
+		s.validateAssumeRoleDuration(),
+	)
+}
+
+func (s Settings) validateAssumeRoleDuration() error {
+	if s.AssumeRoleDuration == 0 {
```

Comment:
I think just checking the minimum here should probably be enough. This won't be 0 unless someone explicitly sets it to 0 due to our defaults.